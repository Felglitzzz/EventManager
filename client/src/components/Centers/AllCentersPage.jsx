import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Centers from './Centers';
// import history from '../../helpers/history';
import { loadCenters } from '../../actions/centerActions';

/**
 *class AllCentersPage
 */
class AllCentersPage extends React.Component {
  /**
   * constructor function
   * @param {object} props
   */
  constructor(props) {
    super(props);

    this.state = {
      centers: []
    };
  }
  /**
   * component lifecycle componentDidMount
   * @returns {void}
   */
  componentDidMount() {
    this.props.loadCenters();
  }

  /**
   * componentWillreceiveProps lifecycle
   *
   * @param {object} nextProps
   *
   * @returns {object} center
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.centers.Centers) {
      this.setState({
        centers: nextProps.centers.Centers
      });
    }
  }
  // /**
  //  * redirctToEdit function
  //  * @returns {void}
  //  */
  // redirectToEdit() {
  //   history.push('/dashboard/centers/:centerId');
  // }
  /**
   * @returns { react } dashboard component
   */
  render() {
    return (
      < Centers
        centers = {this.state.centers}
        redirectToEdit = {this.redirectToEdit}
      />
    );
  }
}

AllCentersPage.propTypes = {
  centers: PropTypes.object.isRequired,
  loadCenters: PropTypes.func.isRequired
};


/**
 * @param {object} state
 * @param {object} ownProps
 * @returns {object} loadedcenters
 */
const mapStateToProps = state => ({
  centers: state.centers.loadedCenters,
});

const mapDispatchToProps = dispatch => ({
  loadCenters: () => dispatch(loadCenters())
});

export default connect(mapStateToProps, mapDispatchToProps)(AllCentersPage);

