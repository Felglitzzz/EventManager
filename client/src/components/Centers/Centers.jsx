import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Loader from 'react-md-spinner';
/**
 * class Centers
 * @returns {object} react component
 */
const Centers = ({ centers }) => (
  centers.length === 0
    ?
    <div className="d-flex justify-content-center pad">
      <Loader
        size={96}
        color1="#f6682f"
        color2="#f6682f"
        color3="#f6682f"
        color4="#f6682f"/>
    </div>
    :
    centers.map((center, id) => {
      localStorage.setItem('centers', JSON.stringify(centers));
      return (

        <div className="col-sm-12 col-md-6 col-lg-4" key={id}>
          <div className="card-deck">
            <div className="card">
              <div className="img-fluid">
                <img
                  className="img-fluid"
                  width="400"
                  height="264"
                  src= {center.image}
                  alt={center.name}/>
              </div>
              <div className="card-body">
                <ul className="list-group list-group-flush">
                  <li className="list-group-item"><h4 className="card-title">{center.name}</h4></li>
                  <li className="list-group-item"><
                    p className="card-text">{center.location}</p></li>
                  <li className="list-group-item d-flex">
                    <p className="card-text">{center.capacity} capacity</p>
                    <p className=" ml-auto card-text">$ {center.price}</p>
                  </li>
                </ul>
              </div>
              <div className="overlay">
                <div className="user-icons">
                  <div className="container pt-2 px-0">
                    <ul className="pl-0">
                      <li className="d-flex justify-content-center">
                        <Link to={`/dashboard/centers/edit/${center.id}`}>
                          <p className="text-orange border-2 border-orange icon-lg twitter mr-1">
                            <i className="fa fa-pencil fa-2x p-2"
                              aria-hidden="true" /></p>
                        </Link>
                        <a className="text-orange border-2 border-orange icon-lg ig mr-1">
                          <i className="fa fa-trash fa-2x p-2"
                            aria-hidden="true" />
                        </a>
                        <Link to ={`/dashboard/center/view/${center.id}`}>
                          <p
                            className="text-orange border-2 border-orange icon-lg twitter mr-1" >
                            <i className="fa fa-eye fa-2x p-2"
                              aria-hidden="true" />
                          </p>
                        </Link>
                        {/* <a>
                          <p
                            className="text-orange border-2 border-orange icon-lg ig mr-1"
                            onClick={handleDelete}
                            id={event.id}
                          >
                            <i className="fa fa-trash fa-2x p-2" id={event.id} aria-hidden="true" />
                          </p>
                        </a> */}
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      );
    })
);
export default Centers;

Centers.propTypes = {
  centers: PropTypes.array.isRequired
};

