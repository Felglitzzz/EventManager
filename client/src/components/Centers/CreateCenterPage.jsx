import React from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';

import PropTypes from 'prop-types';
import history from '../../helpers/history';

import { createCenter } from '../../actions/centerActions';
import { uploadToCloudinary } from '../../actions/imageActions';
import CenterForm from './Form/CenterForm';
import Validate from '../../helpers/validations/Validate';
/**
 * class CreateCenterPage
 */
class CreateCenterPage extends React.Component {
  /**
   * constructor function
   * @param {object} props
   */
  constructor(props) {
    super(props);

    this.state = {
      centerData: {
        name: '',
        location: '',
        capacity: '',
        price: '',
        facilities: [],
        type: '',
        image: '',
      },
      chosenImage: '',
      uploadHeight: '',
      uploadWidth: '',
      errors: {},
      options: [],
      isLoading: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.redirectToCenters = this.redirectToCenters.bind(this);
    this.imageOnChange = this.imageOnChange.bind(this);
    this.selectOnChange = this.selectOnChange.bind(this);
  }

  /**
   * componentWillreceiveProps lifecycle
   * @param {object} nextProps
   * @returns {object} event
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.imageUrl) {
      this.setState({
        centerData: {
          ...this.state.centerData,
          image: nextProps.imageUrl.secure_url
        }
      });
    }
  }

  /**
   * onchange event function
   * @param {*} event
   * @returns {object} new state
   */
  onChange(event) {
    event.persist();
    const field = event.target.name;
    const { centerData } = this.state;
    centerData[field] = event.target.value;
    return this.setState({ centerData });
  }

  /**
   * onchange event function
   * @param {*} event
   * @returns {object} new state
   */
  selectOnChange(event) {
    event.persist();
    if (event.target.checked) {
      return this.setState({
        centerData: {
          ...this.state.centerData,
          facilities: this.state.centerData.facilities.concat(event.target.value)
        }
      });
    }
    const { facilities } = this.state.centerData;
    const name = event.target.value;
    for (let i = facilities.length - 1; i >= 0; i -= 1) {
      if (facilities[i] === name) {
        facilities.splice(i, 1);
        this.setState({
          centerData: {
            ...this.state.centerData,
            facilities
          }
        });
      }
    }

    return this.setState({
      centerData: {
        ...this.state.centerData,
        facilities: this.state.centerData.facilities.slice()
      }
    });
  }

  /**
   * @description handler for image upload - imageOnChange
   * @param {object} event
   * @returns {object} selected file
   */
  imageOnChange(event) {
    const chosenImage = event.target.files[0];
    const imageReader = new FileReader();
    if (chosenImage) {
      imageReader.onload = () => {
        const upload = new Image();
        upload.src = imageReader.result;
        upload.onload = () => {
          this.setState({
            uploadHeight: upload.height,
            uploadWidth: upload.width,
            chosenImage
          });
        };
      };
    }
    imageReader.readAsDataURL(chosenImage);
  }

  /**
   * @description handles on focus event
   * @method handleOnFocus
   *
   * @param { object } event object
   *
   * @returns { object } empty string
   */
  handleFocus(event) {
    const field = event.target.name;
    const { errors } = this.state;
    errors[field] = '';
    this.setState({
      errors
    });
  }

  /**
   *@returns {void}
   */
  validate() {
    const { chosenImage, centerData } = this.state;
    const { errors, isValid } = Validate.createCenter(centerData, chosenImage);
    if (!isValid) {
      this.setState({ errors, isLoading: false });
      return errors;
    }
    this.setState({ errors: {}, isLoading: true });
  }

  /**
   * onSubmit event function
   * @param {*} event
   * @returns {void}
   */
  onSubmit(event) {
    event.persist();
    event.preventDefault();
    const errors = this.validate();
    if (errors) return;
    const { chosenImage } = this.state;
    this.props.uploadToCloudinary(chosenImage)
      .then(() => {
        const { centerData } = this.state;
        if (centerData.image) {
          this.props.createCenter(centerData)
            .then(() => this.redirectToCenters())
            .catch((error) => {
              this.setState({ isLoading: false });
              toastr.error(error);
            });
        }
      })
      .catch(() => {
        this.setState({ isLoading: false });
      });
  }

  /**
   * @param {string} status
   * @returns {void}
   */
  redirectToCenters() {
    this.setState({ isLoading: false });
    toastr.success('Center Created');
    history.replace('/dashboard/centers');
  }

  /**
  * @returns { react } component
  */
  render() {
    return (
      <div>
        <CenterForm
          onSubmit={this.onSubmit}
          onChange={this.onChange}
          centerData={this.state.centerData}
          imageOnChange={this.imageOnChange}
          errors={this.state.errors}
          isLoading={this.state.isLoading}
          handleFocus={this.handleFocus}
          selectOnChange={this.selectOnChange}
        />
      </div>
    );
  }
}
CreateCenterPage.propTypes = {
  createCenter: PropTypes.func.isRequired,
  uploadToCloudinary: PropTypes.func.isRequired,
  imageUrl: PropTypes.object,
};

/**
 * @param {object} state
 * @returns {object} state
 */
function mapStateToProps(state) {
  return {
    imageUrl: state.images.image
  };
}

/**
 *
 * @param {func} dispatch
 * @returns {object} action
 */
function mapDispatchToProps(dispatch) {
  return {
    createCenter: centerData => dispatch(createCenter(centerData)),
    uploadToCloudinary: image => dispatch(uploadToCloudinary(image))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateCenterPage);
