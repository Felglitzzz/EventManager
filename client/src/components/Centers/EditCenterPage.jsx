import React from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import PropTypes from 'prop-types';
import history from '../../helpers/history';

import { updateCenter } from '../../actions/centerActions';
import EditCenterForm from './Form/EditCenterForm';
/**
 * class EditCenterPage
 */
class EditCenterPage extends React.Component {
  /**
   * constructor function
   * @param {object} props
   */
  constructor(props) {
    super(props);

    this.state = {
      updateCenterData: { ...this.props.updateCenterData },
      errors: {},
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.redirectToCenters = this.redirectToCenters.bind(this);
  }

  /**
   * componentWillreceiveProps lifecycle
   * @param {object} nextProps
   * @returns {object} event
   */
  componentWillReceiveProps(nextProps) {
    if (this.props.updateCenterData.id !== nextProps.updateCenterData.id) {
      this.setState({
        updateCenterData: nextProps.updateCenterData
      });
    }
  }

  /**
   * onchange event function
   * @param {*} event
   * @returns {object} new state
   */
  onChange(event) {
    event.persist();
    const field = event.target.name;
    const temp = { ...this.state.updateCenterData };
    temp[field] = event.target.value;
    return this.setState({ updateCenterData: temp });
  }

  /**
   * onSubmit event function
   * @param {*} event
   * @returns {void}
   */
  onSubmit(event) {
    event.persist();
    event.preventDefault();
    const { updateCenterData } = this.state;
    this.props.updateCenter(updateCenterData)
      .then(() => this.redirectToCenters())
      .catch((error) => {
        toastr.error(error);
      });
  }

  /**
   * @returns {void}
   */
  redirectToCenters() {
    this.setState({ isLoading: false });
    toastr.success('Center updated');
    history.push('/dashboard/centers');
  }

  /**
  * @returns { react } component
  */
  render() {
    return (
      <div>
        <EditCenterForm
          onSubmit={this.onSubmit}
          onChange={this.onChange}
          updateCenterData={this.state.updateCenterData}
        />
      </div>
    );
  }
}
EditCenterPage.propTypes = {
  updateCenter: PropTypes.func.isRequired,
  updateCenterData: PropTypes.object.isRequired,
};

const getCenterById = (centers, id) => {
  const centerForUpdate = centers.filter(center => center.id === id);
  if (centerForUpdate.length) return centerForUpdate[0];
  return null;
};
/**
 *
 * @param {object} state
 * @param {object} ownProps
 * @returns {object} state
 */
function mapStateToProps(state, ownProps) {
  const centerId = parseInt(ownProps.match.params.centerId, 10);

  let updateCenterData = {
    id: '', name: '', centerId: '', date: '', time: '', description: '', image: ''
  };

  const centers = state.centers.loadedCenters.Centers;
  if (centerId && centers.length > 0) {
    updateCenterData = getCenterById(centers, centerId);
  }
  return {
    updateCenterData
  };
}

/**
 *
 * @param {func} dispatch
 * @param {object} ownProps
 * @returns {object} action
 */
function mapDispatchToProps(dispatch) {
  return {
    updateCenter: updateCenterData => dispatch(updateCenter(updateCenterData)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditCenterPage);
