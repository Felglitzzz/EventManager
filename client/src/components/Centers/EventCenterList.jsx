import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Loader from 'react-md-spinner';
import getUserFromToken from '../../utils/getUserFromToken';

const EventCenterList = ({
  events, handleCancelEvent, handleApproveEvent, eventLoading, showLoader, error,
  approveEventLoading, cancelEventLoading, eventId, showNoEvents
}) => {
  const { isAdmin } = getUserFromToken();
  events = events.rows;
  if (eventLoading) {
    return showLoader();
  }
  if (error === 'Event Not Found!') {
    return showNoEvents();
  }
  return (
    <div className="p-3">
      <div className="row border border-top-2 border-bottom-2 py-2">
        <div className={isAdmin ? 'col-3' : 'col-4'}>
          <p className="mt-3 font-weight-bold">Date Booked</p></div>
        <div className={isAdmin ? 'col-3' : 'col-4'}>
          <p className="mt-3 font-weight-bold">Status</p></div>
        <div className={isAdmin ? 'col-3' : 'col-4'}>
          <p className="mt-3 font-weight-bold">Events Slated</p></div>
        {isAdmin ?
          <div className="col-3">
            <p className="mt-3 font-weight-bold">Admin Actions</p></div>
          : (null)}
      </div>
      {events.map((event, id) => (
        <div className="row border border-top-2 border-bottom-2 py-2"
          key={id}>
          <div className={isAdmin ? 'col-3' : 'col-4'}><p className="mt-3">{moment(event.startDate).format('dddd, MMMM Do YYYY')}</p></div>
          <div className={isAdmin ? 'col-3 mt-3' : 'col-4 mt-3'}>
            <span className="fa-stack">
              { event.status === 'pending' ?
                <i className="fa fa-circle fa-stack text-orange" />
                : event.status === 'cancelled' ?
                  <i className="fa fa-circle fa-stack text-danger" />
                  : <i className="fa fa-circle fa-stack text-success" />
              }
            </span>{event.status}</div>
          <div className={isAdmin ? 'col-3' : 'col-4'}><p className="mt-3">{event.name}</p></div>
          {isAdmin ?
            <div className="col-3">
              <button
                className="btn-success mt-3 mr-1 w-60"
                id={event.id}
                name={event.name}
                onClick = {handleApproveEvent}>
                {approveEventLoading && +event.id === +eventId ?
                  <Loader
                    color1="#ffffff"
                    color2="#ffffff"
                    color3="#ffffff"
                    color4="#ffffff"
                    size={20} />
                  :
                  'Accept'
                }
              </button>
              <button
                className="btn-danger w-60"
                id={event.id}
                name={event.name}
                onClick = {handleCancelEvent}>
                {cancelEventLoading && +event.id === +eventId ?
                  <Loader
                    color1="#ffffff"
                    color2="#ffffff"
                    color3="#ffffff"
                    color4="#ffffff"
                    size={20} />
                  :
                  'Cancel'
                }</button>
            </div>
            : (null)}
        </div>
      ))}
    </div>
  );
};

EventCenterList.propTypes = {
  events: PropTypes.object.isRequired,
  handleCancelEvent: PropTypes.func.isRequired,
  handleApproveEvent: PropTypes.func.isRequired,
  cancelEventLoading: PropTypes.bool.isRequired,
  approveEventLoading: PropTypes.bool.isRequired,
  showNoEvents: PropTypes.func.isRequired,
  showLoader: PropTypes.func.sRequired,
  eventLoading: PropTypes.bool.isrequired,
  error: PropTypes.object.isRequired,
  eventId: PropTypes.number.isrequired
};

export default EventCenterList;
