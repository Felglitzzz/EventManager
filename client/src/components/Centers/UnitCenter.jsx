import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import getUserFromToken from '../../utils/getUserFromToken';

/**
 * @description - Stateless component for rendering a center
 *
 * @returns {jsx} UnitCenter - Rendered view
 */
const UnitCenter = ({ centers }) => {
  const { isAdmin } = getUserFromToken();
  centers = centers.rows;
  return (
    centers.map((center, id) => (
      <div className="col-sm-12 col-md-6 col-lg-4 my-2"
        key={id}>
        <div className="card-deck">
          <div className="card">
            <div className="img-fluid deck">
              <img
                alt={center.name}
                className="img-fluid deck"
                height="264"
                src= {center.image}
                width="400" />
            </div>
            <div className="card-body p-2">
              <ul className="list-group list-group-flush">
                <li className="list-group-item px-2 py-2"><h4
                  className="card-title">{center.name}</h4></li>
                <li className="list-group-item pt-0 pb-2 px-2 border-top-0"><p
                  className="card-text">{center.location}</p></li>
                <li className="list-group-item d-flex px-2 py-0">
                  <p className="card-text">{center.capacity} capacity</p>
                  <p className=" ml-auto card-text">NGN {center.price}</p>
                </li>
              </ul>
            </div>
            <div className="overlay">
              <div className="user-icons">
                <div className="container pt-2 px-0">
                  <ul className="pl-0">
                    <li className="d-flex justify-content-center">
                      { isAdmin ?
                        <Link to={`/dashboard/centers/${center.id}`}>
                          <p className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim">
                            <i aria-hidden="true"
                              className="fa fa-pencil fa-2x p-2" /></p>
                        </Link> :
                        (null)
                      }
                      <Link
                        to ={`/dashboard/centers/view/${center.id}`}>
                        <p
                          className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim" >
                          <i aria-hidden="true"
                            className="fa fa-eye fa-2x p-2" />
                        </p>
                      </Link>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    ))
  );
};
export default UnitCenter;

UnitCenter.propTypes = {
  centers: PropTypes.object.isRequired
};

