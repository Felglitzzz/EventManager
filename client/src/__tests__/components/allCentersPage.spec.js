import React from 'react';
import { shallow, mount } from 'enzyme';

import { AllCentersPage, mapDispatchToProps } from '../../components/Centers/AllCentersPage';
import UnitCenter from '../../components/Centers/UnitCenter';
import mockLocalStorage from '../__mocks__/mockLocalStorage';

window.localStorage = mockLocalStorage;


const props = {
  loadCenters: () => Promise.resolve(),
  centers: {
    centers: {
      centers: {
        count: 13,
        rows: [
          {
            id: 15,
            name: 'jkddkd',
            location: 'Alausa',
            capacity: 900,
            price: 900,
            facilities: [
              'Casino',
              'Parking Space',
              'Volley Ball Court',
              'Salon and Spa',
              'Basket Ball Court'
            ],
            type: 'jherlfrihn',
            description: 'bnmd',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
            createdAt: '2018-06-10',
            updatedAt: '2018-06-10'
          }
        ]
      },
      meta: {
        pagination: {
          currentPageUrl: 'http://localhost:1991/api/v1/centers?page=1',
          next: 'http://localhost:1991/api/v1/centers?page=2',
          currentPage: 1,
          totalPages: 5,
          offset: 0,
          limit: 3
        }
      }
    }
  },
  history: {
    push: jest.fn(),
  },
  match: {
    params: {},
    path: '/dashboard/events'
  },
  location: {
    search: '?page=1'
  },
};

jest.mock('react-router-dom');


describe(' All Center\'s Page', () => {
  it('should render all centers component', () => {
    const wrapper = shallow(<AllCentersPage {...props} />);
    expect(wrapper.find('section').exists).toBeTruthy();
  });

  it('should render stateless child component', () => {
    const childProps = {
      loadCenters: () => Promise.resolve(),
      centers: {
        count: 13,
        rows: [
          {
            id: 15,
            name: 'jkddkd',
            location: 'Alausa',
            capacity: 900,
            price: 900,
            facilities: [
              'Casino',
              'Parking Space',
              'Volley Ball Court',
              'Salon and Spa',
              'Basket Ball Court'
            ],
            type: 'jherlfrihn',
            description: 'bnmd',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
            createdAt: '2018-06-10',
            updatedAt: '2018-06-10'
          }
        ]
      }
    };
    const wrapper = shallow(<UnitCenter {...childProps} />);
    wrapper.setProps({
      centers: {
        count: 13,
        rows: [
          {
            id: 15,
            name: 'jkddkd',
            location: 'Alausa',
            capacity: 900,
            price: 900,
            facilities: [
              'Casino',
              'Parking Space',
              'Volley Ball Court',
              'Salon and Spa',
              'Basket Ball Court'
            ],
            type: 'jherlfrihn',
            description: 'bnmd',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
            createdAt: '2018-06-10',
            updatedAt: '2018-06-10'
          }
        ]
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should test componentWillReceiveProps method', () => {
    const nextProps = {
      loadAllEvent: () => Promise.resolve(),
      centers: {
        centers: {
          centers: {
            count: 13,
            rows: [
              {
                id: 15,
                name: 'jkddkd',
                location: 'Alausa',
                capacity: 900,
                price: 900,
                facilities: [
                  'Casino',
                  'Parking Space',
                  'Volley Ball Court',
                  'Salon and Spa',
                  'Basket Ball Court'
                ],
                type: 'jherlfrihn',
                description: 'bnmd',
                image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
                createdAt: '2018-06-10',
                updatedAt: '2018-06-10'
              }
            ]
          },
          meta: {
            pagination: {
              currentPageUrl: 'http://localhost:1991/api/v1/centers?page=1',
              next: 'http://localhost:1991/api/v1/centers?page=2',
              currentPage: 1,
              totalPages: 5,
              offset: 0,
              limit: 3
            }
          }
        }
      },
      history: {
        push: jest.fn(),
      },
      match: {
        params: {}
      },
      location: {
        search: '?page=1'
      }
    };
    const componentWillReceivePropsSpy = jest.spyOn(
      AllCentersPage.prototype,
      'componentWillReceiveProps'
    );
    const wrapper = shallow(<AllCentersPage {...props} />);
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
  });

  it('should test mapDispatchToProps method', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).loadCenters).toBeTruthy();
  });

  it('should test showNext method', () => {
    const history = { push: () => {} };

    const showNextSpy = jest.spyOn(AllCentersPage.prototype, 'showNext');
    const wrapper = mount(<AllCentersPage
      {...props}
      history = {history}
    />);
    wrapper.setState({
      pagination: {
        currentPageUrl: 'http://localhost:1991/api/v1/centers?page=1',
        next: 'http://localhost:1991/api/v1/centers?page=2',
        currentPage: 1,
        totalPages: 5,
        offset: 0,
        limit: 3
      },
      centersLoading: false,
      centers: {
        count: 13,
        rows: [
          {
            id: 15,
            name: 'jkddkd',
            location: 'Alausa',
            capacity: 900,
            price: 900,
            facilities: [
              'Casino',
              'Parking Space',
              'Volley Ball Court',
              'Salon and Spa',
              'Basket Ball Court'
            ],
            type: 'jherlfrihn',
            description: 'bnmd',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
            createdAt: '2018-06-10',
            updatedAt: '2018-06-10'
          }
        ]
      },
    });
    wrapper.find('#showNext').simulate('click');
    expect(showNextSpy).toHaveBeenCalled();
  });

  it('should test showPrevious method', () => {
    const history = { push: () => {} };

    const showPreviousSpy = jest.spyOn(AllCentersPage.prototype, 'showPrevious');
    const wrapper = mount(<AllCentersPage
      {...props}
      history = {history}
    />);
    wrapper.setState({
      pagination: {
        previous: 'http://localhost:1991/api/v1/centers?page=1',
        currentPageUrl: 'http://localhost:1991/api/v1/centers?page=2',
        next: 'http://localhost:1991/api/v1/centers?page=3',
        currentPage: 2,
        totalPages: 5,
        offset: 0,
        limit: 3
      },
      centersLoading: false,
      centers: {
        count: 13,
        rows: [
          {
            id: 15,
            name: 'jkddkd',
            location: 'Alausa',
            capacity: 900,
            price: 900,
            facilities: [
              'Casino',
              'Parking Space',
              'Volley Ball Court',
              'Salon and Spa',
              'Basket Ball Court'
            ],
            type: 'jherlfrihn',
            description: 'bnmd',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg',
            createdAt: '2018-06-10',
            updatedAt: '2018-06-10'
          }
        ]
      },
    });
    wrapper.find('#showPrevious').simulate('click');
    expect(showPreviousSpy).toHaveBeenCalled();
  });

  it('should show no event page if there is no events', () => {
    const showNoCentersSpy = jest.spyOn(AllCentersPage.prototype, 'showNoCenters');
    const wrapper = shallow(<AllCentersPage {...props} />);

    wrapper.setState({
      centersLoading: false,
      centers: {}
    });

    const instance = wrapper.instance();
    instance.showNoCenters();
    expect(showNoCentersSpy).toHaveBeenCalled();
  });
});

