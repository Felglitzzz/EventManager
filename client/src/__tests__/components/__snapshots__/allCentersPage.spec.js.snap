// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` All Center's Page should render stateless child component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UnitCenter
    centers={
      Object {
        "count": 13,
        "rows": Array [
          Object {
            "capacity": 900,
            "createdAt": "2018-06-10",
            "description": "bnmd",
            "facilities": Array [
              "Casino",
              "Parking Space",
              "Volley Ball Court",
              "Salon and Spa",
              "Basket Ball Court",
            ],
            "id": 15,
            "image": "https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg",
            "location": "Alausa",
            "name": "jkddkd",
            "price": 900,
            "type": "jherlfrihn",
            "updatedAt": "2018-06-10",
          },
        ],
      }
    }
    loadCenters={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      className="col-sm-12 col-md-6 col-lg-4 my-2"
    >
      <div
        className="card-deck"
      >
        <div
          className="card"
        >
          <div
            className="img-fluid deck"
          >
            <img
              alt="jkddkd"
              className="img-fluid deck"
              height="264"
              src="https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg"
              width="400"
            />
          </div>
          <div
            className="card-body p-2"
          >
            <ul
              className="list-group list-group-flush"
            >
              <li
                className="list-group-item px-2 py-2"
              >
                <h4
                  className="card-title"
                >
                  jkddkd
                </h4>
              </li>
              <li
                className="list-group-item pt-0 pb-2 px-2 border-top-0"
              >
                <p
                  className="card-text"
                >
                  Alausa
                </p>
              </li>
              <li
                className="list-group-item d-flex px-2 py-0"
              >
                <p
                  className="card-text"
                >
                  900
                   capacity
                </p>
                <p
                  className=" ml-auto card-text"
                >
                  NGN 
                  900
                </p>
              </li>
            </ul>
          </div>
          <div
            className="overlay"
          >
            <div
              className="user-icons"
            >
              <div
                className="container pt-2 px-0"
              >
                <ul
                  className="pl-0"
                >
                  <li
                    className="d-flex justify-content-center"
                  >
                    <Link
                      id="viewCenterButton"
                      replace={false}
                      to="/dashboard/centers/view/15"
                    >
                      <p
                        className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-eye fa-2x p-2"
                        />
                      </p>
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        className="col-sm-12 col-md-6 col-lg-4 my-2"
      >
        <div
          className="card-deck"
        >
          <div
            className="card"
          >
            <div
              className="img-fluid deck"
            >
              <img
                alt="jkddkd"
                className="img-fluid deck"
                height="264"
                src="https://res.cloudinary.com/felglitz/image/upload/v1528623443/no30px0yjsdbzaddtcch.jpg"
                width="400"
              />
            </div>
            <div
              className="card-body p-2"
            >
              <ul
                className="list-group list-group-flush"
              >
                <li
                  className="list-group-item px-2 py-2"
                >
                  <h4
                    className="card-title"
                  >
                    jkddkd
                  </h4>
                </li>
                <li
                  className="list-group-item pt-0 pb-2 px-2 border-top-0"
                >
                  <p
                    className="card-text"
                  >
                    Alausa
                  </p>
                </li>
                <li
                  className="list-group-item d-flex px-2 py-0"
                >
                  <p
                    className="card-text"
                  >
                    900
                     capacity
                  </p>
                  <p
                    className=" ml-auto card-text"
                  >
                    NGN 
                    900
                  </p>
                </li>
              </ul>
            </div>
            <div
              className="overlay"
            >
              <div
                className="user-icons"
              >
                <div
                  className="container pt-2 px-0"
                >
                  <ul
                    className="pl-0"
                  >
                    <li
                      className="d-flex justify-content-center"
                    >
                      <Link
                        id="viewCenterButton"
                        replace={false}
                        to="/dashboard/centers/view/15"
                      >
                        <p
                          className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-eye fa-2x p-2"
                          />
                        </p>
                      </Link>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
